### ISSUE TITLE
Prune does not delete unused images

### ISSUE THREAD

**Issue by rfrowe** [swipely/docker-api#506]
As per the Docker API, `/images/prune` should remove all unused images. However `Docker::Images.prune` never removes unused images. I can only ever get it to delete dangling images.

Using Docker Engine 17.09, I have tried:
```
Docker::Image.prune
Docker.connection.post('/images/prune', {all: true})
Docker.connection.post('/images/prune', {filters: {dangling: ['true']}.to_json})
Docker.connection.post('/images/prune', {filters: {dangling: ['false']}.to_json})
```

None of them will remove unused images. I pulled several images that were never used by any containers and they were not removed.

Either this is an issue with this gem or the Docker API itself. Can anyone else reproduce this issue?

**Comment by tlunter** [comment]
In this scenario, that's all we're doing: https://github.com/swipely/docker-api/blob/master/lib/docker/image.rb#L140-L143

Seems like there must be an issue with the daemon itself.

Can dig into here through the versions and see if something changed there: https://github.com/moby/moby/blob/master/daemon/prune.go#L164-L313

**Comment by rfrowe** [comment]
@tlunter I'm really not sure. I ran the same exact command:

`Docker.connection.post('/images/prune', {filters: {dangling: ['false']}.to_json})`

today and it worked, so I have no idea. Either way, I opened #507 to allow options to be specified so you don't have to manually post the request.

**Comment by rfrowe** [comment]
@tlunter Reason for closure? If this is solved, great! If this is won't fix, you should close #507. I don't care either way, as thankfully I'm not working on the same thing 2 years later :)
